daemon/main.go

```diff
@@ -0,0 +1,64 @@
+package main
+
+import (
+	"errors"
+	"flag"
+
+	// Import all Kubernetes client auth plugins (e.g. Azure, GCP, OIDC, etc.)
+	// to ensure that exec-entrypoint and run can make use of them.
+	"go.uber.org/zap/zapcore"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+	//+kubebuilder:scaffold:imports
+)
+
+var (
+	scheme   = runtime.NewScheme()
+	setupLog = ctrl.Log.WithName("setup")
+)
+
+type Daemon interface {
+	Start()
+}
+
+func isHostMode(mode string) bool {
+	return mode == "host"
+}
+
+func isDpuMode(mode string) bool {
+	return mode == "dpu"
+}
+
+func createDaemon(dev bool, mode string) (Daemon, error) {
+	vsp := NewDummyPlugin(mode)
+	if isHostMode(mode) {
+		return NewHostDaemon(dev, vsp), nil
+	} else if isDpuMode(mode) { 
+		return NewDpuDaemon(dev, vsp), nil
+	} else {
+		return nil, errors.New("Invalid mode")
+	}
+}
+
+func main() {
+	var mode string
+	flag.StringVar(&mode, "mode", "", "Mode for the daemon, can be either host or dpu")
+	dev := flag.Bool("dev", false, "Enable development mode")
+	opts := zap.Options{
+		Development: true,
+		Level:       zapcore.DebugLevel,
+	}
+	opts.BindFlags(flag.CommandLine)
+	flag.Parse()
+	ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))
+
+	log := ctrl.Log.WithName("Daemon Init")
+	log.Info("Daemon init")
+	daemon, err := createDaemon(*dev, mode)
+	if err != nil {
+		log.Error(err, "Failed to start daemon")
```

Comment:
`Failed to create daemon`. We probably want to check on error for start daemon below.

Reply:
please elaborate.

Reply:
The first log message for "createDaemon" is not consistent with the name of the function being called and verified.

I think Start daemon, there will be error/failure cases and we should handle the error for that function. I don't think Start() is a returnless function or is it?

Reply:
Let's address this later. Not an issue right now.